// styled-bootstrap-mixins/dist/util.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.div = exports.mul = exports.sub = exports.add = exports.percentage = exports.ceil = exports.floor = undefined;

var _lodash = require('lodash.defaultto');

var _lodash2 = _interopRequireDefault(_lodash);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var valueOf = function valueOf(numUnit) {
  return parseFloat(numUnit);
};
var unitOf = function unitOf(numUnit) {
  var hasDigit = /\d+/.test(numUnit);
  var output = /[a-z]+/.exec(numUnit);
  if (!output || !hasDigit) {
    return null;
  }
  return output[0];
};

var floor = exports.floor = function floor(number) {
  return '' + (0, _lodash2.default)(Math.floor(valueOf(number)), '') + (0, _lodash2.default)(unitOf(number), '');
};
var ceil = exports.ceil = function ceil(number) {
  return '' + (0, _lodash2.default)(Math.ceil(valueOf(number)), '') + (0, _lodash2.default)(unitOf(number), '');
};
var percentage = exports.percentage = function percentage(number) {
  var output = valueOf(number) * 100;
  if (isNaN(output)) {
    return '';
  }
  return output + '%';
};
var add = exports.add = function add(n1, n2) {
  return '' + (0, _lodash2.default)(valueOf(n1) + valueOf(n2), '') + (0, _lodash2.default)(unitOf(n1) || unitOf(n2), '');
};
var sub = exports.sub = function sub(n1, n2) {
  return '' + (0, _lodash2.default)(valueOf(n1) - valueOf(n2), '') + (0, _lodash2.default)(unitOf(n1) || unitOf(n2), '');
};
var mul = exports.mul = function mul(n1, n2) {
  return '' + (0, _lodash2.default)(valueOf(n1) * valueOf(n2), '') + (0, _lodash2.default)(unitOf(n1) || unitOf(n2), '');
};
var div = exports.div = function div(n1, n2) {
  var _n1 = valueOf(n1);
  var _n2 = valueOf(n2);

  if (_n1 === 0 && _n2 === 0) {
    return '';
  }
  return '' + (0, _lodash2.default)(valueOf(n1) / valueOf(n2), '') + (0, _lodash2.default)(unitOf(n1) || unitOf(n2), '');
};
