// styled-bootstrap-mixins/dist/mixins/grid.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.make_lg_column_pull = exports.make_lg_column_push = exports.make_lg_column_offset = exports.make_lg_column = exports.make_md_column_pull = exports.make_md_column_push = exports.make_md_column_offset = exports.make_md_column = exports.make_sm_column_pull = exports.make_sm_column_push = exports.make_sm_column_offset = exports.make_sm_column = exports.make_xs_column_pull = exports.make_xs_column_push = exports.make_xs_column_offset = exports.make_xs_column = exports.make_row = exports.container_fixed = undefined;

var _templateObject = _taggedTemplateLiteral(['\n  margin-right: auto;\n  margin-left: auto;\n  padding-left:  ', ';\n  padding-right: ', ';\n  ', '\n'], ['\n  margin-right: auto;\n  margin-left: auto;\n  padding-left:  ', ';\n  padding-right: ', ';\n  ', '\n']),
    _templateObject2 = _taggedTemplateLiteral(['\n  margin-left:  ', ';\n  margin-right: ', ';\n  ', '\n'], ['\n  margin-left:  ', ';\n  margin-right: ', ';\n  ', '\n']),
    _templateObject3 = _taggedTemplateLiteral(['\n  position: relative;\n  float: left;\n  width: ', ';\n  min-height: 1px;\n  padding-left:  ', ';\n  padding-right: ', ';\n'], ['\n  position: relative;\n  float: left;\n  width: ', ';\n  min-height: 1px;\n  padding-left:  ', ';\n  padding-right: ', ';\n']),
    _templateObject4 = _taggedTemplateLiteral(['\n  margin-left: ', ';\n'], ['\n  margin-left: ', ';\n']),
    _templateObject5 = _taggedTemplateLiteral(['\n  left: ', ';\n'], ['\n  left: ', ';\n']),
    _templateObject6 = _taggedTemplateLiteral(['\n  right: ', ';\n'], ['\n  right: ', ';\n']),
    _templateObject7 = _taggedTemplateLiteral(['\n  position: relative;\n  min-height: 1px;\n  padding-left:  ', ';\n  padding-right: ', ';\n\n  @media (min-width: ', ') {\n    float: left;\n    width: ', ';\n  }\n'], ['\n  position: relative;\n  min-height: 1px;\n  padding-left:  ', ';\n  padding-right: ', ';\n\n  @media (min-width: ', ') {\n    float: left;\n    width: ', ';\n  }\n']),
    _templateObject8 = _taggedTemplateLiteral(['\n  @media (min-width: ', ') {\n    margin-left: ', ';\n  }\n'], ['\n  @media (min-width: ', ') {\n    margin-left: ', ';\n  }\n']),
    _templateObject9 = _taggedTemplateLiteral(['\n  @media (min-width: ', ') {\n    left: ', ';\n  }\n'], ['\n  @media (min-width: ', ') {\n    left: ', ';\n  }\n']),
    _templateObject10 = _taggedTemplateLiteral(['\n  @media (min-width: ', ') {\n    right: ', ';\n  }\n'], ['\n  @media (min-width: ', ') {\n    right: ', ';\n  }\n']);

var _constants = require('../constants.json');

var _constants2 = _interopRequireDefault(_constants);

var _clearfix = require('./clearfix');

var _styledComponents = require('styled-components');

var _util = require('../util');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var $GRID_GUTTER_WIDTH = _constants2.default['$grid-gutter-width'];
var $GRID_COLUMNS = _constants2.default['$grid-columns'];
var $SCREEN_SM_MIN = _constants2.default['$screen-sm-min'];
var $SCREEN_MD_MIN = _constants2.default['$screen-md-min'];
var $SCREEN_LG_MIN = _constants2.default['$screen-lg-min'];

var container_fixed = exports.container_fixed = function container_fixed() {
  var gutter = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : $GRID_GUTTER_WIDTH;
  return (0, _styledComponents.css)(_templateObject, (0, _util.floor)((0, _util.div)(gutter, 2)), (0, _util.ceil)((0, _util.div)(gutter, 2)), (0, _clearfix.clearfix)());
};

var make_row = exports.make_row = function make_row() {
  var gutter = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : $GRID_GUTTER_WIDTH;
  return (0, _styledComponents.css)(_templateObject2, (0, _util.ceil)((0, _util.div)(gutter, -2)), (0, _util.floor)((0, _util.div)(gutter, -2)), (0, _clearfix.clearfix)());
};

var make_xs_column = exports.make_xs_column = function make_xs_column(columns) {
  var gutter = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : $GRID_GUTTER_WIDTH;
  return (0, _styledComponents.css)(_templateObject3, (0, _util.percentage)((0, _util.div)(columns, $GRID_COLUMNS)), (0, _util.div)(gutter, 2), (0, _util.div)(gutter, 2));
};

var make_xs_column_offset = exports.make_xs_column_offset = function make_xs_column_offset(columns) {
  return (0, _styledComponents.css)(_templateObject4, (0, _util.percentage)((0, _util.div)(columns, $GRID_COLUMNS)));
};

var make_xs_column_push = exports.make_xs_column_push = function make_xs_column_push(columns) {
  return (0, _styledComponents.css)(_templateObject5, (0, _util.percentage)((0, _util.div)(columns, $GRID_COLUMNS)));
};

var make_xs_column_pull = exports.make_xs_column_pull = function make_xs_column_pull(columns) {
  return (0, _styledComponents.css)(_templateObject6, (0, _util.percentage)((0, _util.div)(columns, $GRID_COLUMNS)));
};

var make_sm_column = exports.make_sm_column = function make_sm_column(columns) {
  var gutter = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : $GRID_GUTTER_WIDTH;
  return (0, _styledComponents.css)(_templateObject7, (0, _util.div)(gutter, 2), (0, _util.div)(gutter, 2), $SCREEN_SM_MIN, (0, _util.percentage)((0, _util.div)(columns, $GRID_COLUMNS)));
};

var make_sm_column_offset = exports.make_sm_column_offset = function make_sm_column_offset(columns) {
  return (0, _styledComponents.css)(_templateObject8, $SCREEN_SM_MIN, (0, _util.percentage)((0, _util.div)(columns, $GRID_COLUMNS)));
};

var make_sm_column_push = exports.make_sm_column_push = function make_sm_column_push(columns) {
  return (0, _styledComponents.css)(_templateObject9, $SCREEN_SM_MIN, (0, _util.percentage)((0, _util.div)(columns, $GRID_COLUMNS)));
};

var make_sm_column_pull = exports.make_sm_column_pull = function make_sm_column_pull(columns) {
  return (0, _styledComponents.css)(_templateObject10, $SCREEN_SM_MIN, (0, _util.percentage)((0, _util.div)(columns, $GRID_COLUMNS)));
};

var make_md_column = exports.make_md_column = function make_md_column(columns) {
  var gutter = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : $GRID_GUTTER_WIDTH;
  return (0, _styledComponents.css)(_templateObject7, (0, _util.div)(gutter, 2), (0, _util.div)(gutter, 2), $SCREEN_MD_MIN, (0, _util.percentage)((0, _util.div)(columns, $GRID_COLUMNS)));
};

var make_md_column_offset = exports.make_md_column_offset = function make_md_column_offset(columns) {
  return (0, _styledComponents.css)(_templateObject8, $SCREEN_MD_MIN, (0, _util.percentage)((0, _util.div)(columns, $GRID_COLUMNS)));
};

var make_md_column_push = exports.make_md_column_push = function make_md_column_push(columns) {
  return (0, _styledComponents.css)(_templateObject9, $SCREEN_MD_MIN, (0, _util.percentage)((0, _util.div)(columns, $GRID_COLUMNS)));
};

var make_md_column_pull = exports.make_md_column_pull = function make_md_column_pull(columns) {
  return (0, _styledComponents.css)(_templateObject10, $SCREEN_MD_MIN, (0, _util.percentage)((0, _util.div)(columns, $GRID_COLUMNS)));
};

var make_lg_column = exports.make_lg_column = function make_lg_column(columns) {
  var gutter = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : $GRID_GUTTER_WIDTH;
  return (0, _styledComponents.css)(_templateObject7, (0, _util.div)(gutter, 2), (0, _util.div)(gutter, 2), $SCREEN_LG_MIN, (0, _util.percentage)((0, _util.div)(columns, $GRID_COLUMNS)));
};

var make_lg_column_offset = exports.make_lg_column_offset = function make_lg_column_offset(columns) {
  return (0, _styledComponents.css)(_templateObject8, $SCREEN_LG_MIN, (0, _util.percentage)((0, _util.div)(columns, $GRID_COLUMNS)));
};

var make_lg_column_push = exports.make_lg_column_push = function make_lg_column_push(columns) {
  return (0, _styledComponents.css)(_templateObject9, $SCREEN_LG_MIN, (0, _util.percentage)((0, _util.div)(columns, $GRID_COLUMNS)));
};

var make_lg_column_pull = exports.make_lg_column_pull = function make_lg_column_pull(columns) {
  return (0, _styledComponents.css)(_templateObject10, $SCREEN_LG_MIN, (0, _util.percentage)((0, _util.div)(columns, $GRID_COLUMNS)));
};
